loadtests/__init__.py

```diff
@@ -1,54 +1,142 @@
+import json
+import random
+import uuid
+
 from requests.auth import HTTPBasicAuth
 from loads.case import TestCase
 
 
-class TestPOC(TestCase):
+ACTIONS_PERCENTAGE = [
+    ('create', 20),
+    ('update', 50),
+    ('filter_sort', 60),
+    ('read_further', 80),
+    ('mark_as_read', 40),
+    ('create_conflict', 10),
+    ('update_conflict', 10),
+    ('archive', 10),
+    ('delete', 10)
+]
+
+
+class TestBasic(TestCase):
     def __init__(self, *args, **kwargs):
-        super(TestPOC, self).__init__(*args, **kwargs)
+        """Initialization that happens once per user.
+        """
+        super(TestBasic, self).__init__(*args, **kwargs)
+
+        self.random_user = uuid.uuid4().hex
+        self.basic_auth = HTTPBasicAuth(self.random_user, 'secret')
 
-        self.alice_auth = HTTPBasicAuth('alice', 'secret')
-        self.john_auth = HTTPBasicAuth('john', 'secret')
+        # Create at least some records for this user
+        total_records = random.randint(0, 100)
+        for i in range(total_records):
+            self.create()
+
+    def setUp(self):
+        # Pick a random record
+        resp = self.session.get(self.api_url('articles'), auth=self.basic_auth)
+        self.random_record = random.choice(resp.json()['items'])
+        self.random_id = self.random_record['_id']
+        self.random_url = self.api_url('articles/{0}'.format(self.random_id))
+
+    def incr_counter(self, name):
+        hit, user, current_hit, current_user = self.session.loads_status
+        if current_user is None:
+            return
+        super(TestBasic, self).incr_counter(name)
 
     def api_url(self, path):
-        return "{0}/v1/{1}".format(self.server_url, path)
+        return "{0}/v0/{1}".format(self.server_url, path)
 
     def test_all(self):
-        self.test_home()
-        self.test_create_record()
-        self.test_update_status()
-        self.test_list_filtered()
-
-    def test_home(self):
-        res = self.session.get(self.api_url('articles'), auth=self.alice_auth)
-        self.assertEqual(res.status_code, 200)
+        for action, percentage in ACTIONS_PERCENTAGE:
```

Comment:
I am confused here because a single call is going to generate one to nine tests. Is this the intended behavior ?